### PR TITLE
EOS-15436:S3:Reducing the footprint of log files.

### PR DESCRIPTION
**Pull request by SeagateChaDeepak**
**S3 Server changes :**

> Changed Entering to Entry.
> Changed Exiting to Exit.
> Function names will be present in only Entry, Exit, Constructor
> and Destructor Log messages.
> Req ID changed to Req.
> Added Stripped Request id which will be used in place of
> request id.
> Received Request with uri shortened to Req uri.
> Memory pool for unit_size shortened to mempool for size.
> Additional print added to display request id the first time
> along with its stripped counterpart.

**Auth Server changes :**

> "Requested action is -" changed to "Action -".
> "ACL authorization request is validated successfully" changed to "ACL
> authorization success".
> "Sending HTTP Response code" changed to "HTTP Response".
> Added Stripped Request id.
> Removed thread and request id print.

Signed-off-by: Deepak Mahale <deepak.mahale@seagate.com>

### CODE REVIEW

File: auth/server/src/main/java/com/seagates3/authserver/AuthServerGetHandler.java [hunk 1]
```diff
@@ -92,6 +92,15 @@ void run() {
       AuthServerConfig.setReqId(BinaryUtil.getAlphaNumericUUID());
     }
 
+    String stripped_request_id = AuthServerConfig.getReqId();
+    AuthServerConfig.setStripedReqId(stripped_request_id);
```
Comment (sachitanands) [inline auth/server/src/main/java/com/seagates3/authserver/AuthServerGetHandler.java:96]: add this as part of else condition in below code..

File: auth/server/src/main/java/com/seagates3/authserver/AuthServerPostHandler.java [hunk 2]
```diff
@@ -72,6 +72,15 @@ public void run() {
           AuthServerConfig.setReqId(BinaryUtil.getAlphaNumericUUID());
         }
 
+        String stripped_request_id = AuthServerConfig.getReqId();
+        AuthServerConfig.setStripedReqId(stripped_request_id);
```
Comment (sachitanands) [inline auth/server/src/main/java/com/seagates3/authserver/AuthServerPostHandler.java:76]: add this as part of else condition in below code..

File: server/request_object.cc [hunk 3]
```diff
@@ -126,7 +126,11 @@ RequestObject::RequestObject(
 
   S3Uuid uuid;
   request_id = uuid.get_string_uuid();
-  s3_log(S3_LOG_DEBUG, request_id, "Constructor\n");
+  // last 12 characters
+  stripped_request_id = std::string(request_id.end() - 12, request_id.end());
```
Comment (amitwac2608) [inline server/request_object.cc:130]: write a unit test to test that the stripped_request_id will always have 12 chars.

File: server/request_object.cc [hunk 4]
```diff
@@ -216,7 +220,7 @@ void RequestObject::initialise() {
 }
 
 RequestObject::~RequestObject() {
-  s3_log(S3_LOG_DEBUG, request_id, "Destructor\n");
+  s3_log(S3_LOG_DEBUG, request_id, "%s Dtor\n", __func__);
```
Comment (amitwac2608) [inline server/request_object.cc:223]: why to log "Dtor" !! __func__ would log the name of the function, and would be understood from this only that this is destructor !!

File: auth/server/src/main/java/com/seagates3/authserver/AuthServerPostHandler.java (hunk 2 above)
Comment (SeagateChaDeepak) [reply auth/server/src/main/java/com/seagates3/authserver/AuthServerPostHandler.java:76]: This part is required to print the entire request Id the first time its encountered.
The second setstripedreqid is used to set the actual stripped request id.

File: auth/server/src/main/java/com/seagates3/authserver/AuthServerGetHandler.java (hunk 1 above)
Comment (SeagateChaDeepak) [reply auth/server/src/main/java/com/seagates3/authserver/AuthServerGetHandler.java:96]: This part is required to print the entire request Id the first time its encountered.
The second setstripedreqid is used to set the actual stripped request id.

File: server/request_object.cc (hunk 4 above)
Comment (SeagateChaDeepak) [reply server/request_object.cc]: Done.

File: server/request_object.cc (hunk 3 above)
Comment (SeagateChaDeepak) [reply server/request_object.cc:130]: Done

**Review by sachitanands — APPROVED** [review]
AuthServer changes looks good.

**Review by knrajnambiar76 — APPROVED** [review]

**Pull request merged**